core/nodepool/config/config.go

```diff
@@ -275,6 +278,14 @@ func (c ProvidedConfig) NodeLabels() model.NodeLabels {
 	return labels
 }
 
+func (c ProvidedConfig) FeatureGates() model.FeatureGates {
```

Comment:
Should be moved to `ComputedConfig` because it is where all the logic should go, whereas  `ProvidedConfig` is the part of the model.